packages/angular_devkit/build_angular/src/utils/index-file/augment-index-html.ts

```diff
@@ -204,6 +223,10 @@ export async function augmentIndexHtml(
             updateAttribute(tag, 'href', baseHref);
           }
           break;
+        case 'link':
+          if (!foundPreconnect && readAttribute(tag, 'rel') === 'preconnect') {
```

Comment:
Am I reading this right that we don't generate preconnects to the image domain if any preconnect tags are found?  It seems fairly common for there to be unrelated preconnects, e.g. for fonts or GA. Should we have a check more specific to the image domains?

Reply:
Yeah, I did that on purpose originally because I was a little worried about missing almost-duplicates, like if they added a preconnect for `"www.example.com/"` and the preconnect generator adds one for `"www.example.com"`. But after thinking about it a little more, I think that's kind of a corner case, and the failure case of having two preconnects for the same domain in the index.html looks a little weird but wouldn't actually cause any harm.

I went back and added a per-domain check, instead of an overall check.